generated_code/aderdg.py

```diff
@@ -125,13 +125,11 @@ def __init__(self, order, multipleSimulations, matricesDir):
     self.db.update(project2nFaceTo3m)
 
     selectVelocitySpp = np.zeros((self.numberOfQuantities(), 3))
-    selectVelocitySpp[6:9,0:3] = np.eye(3)
+    self.setBottomToIdentity(selectVelocitySpp)
     self.selectVelocity = Tensor('selectVelocity', selectVelocitySpp.shape, selectVelocitySpp, CSCMemoryLayout)
 
     self.selectTractionSpp = np.zeros((self.numberOfQuantities(), 3), dtype=bool)
-    self.selectTractionSpp[0,0] = True
-    self.selectTractionSpp[3,1] = True
-    self.selectTractionSpp[5,2] = True
+    self.setSelectTractionSppToTrue(self.selectTractionSpp)
```

Comment:
Likewise, you could take the `mapToTractions` method here for `self.selectTractionSpp`.

Most likely, you won't even need to convert to booleans afterwards.